File: longest-substring-without-repeating-characters/LongestSubStringWithoutRepeatingCharacters.java
    public int lengthOfLongestSubstring(String s){
        int left = 0;
        int right = 0;
        int longSoFar = 0;
        Set<Character> countChar = new HashSet<>();
        while (left < s.length() && right < s.length()){
            if(!countChar.contains(s.charAt(right))){
                countChar.add(s.charAt(right++));
                longSoFar = Math.max(longSoFar, right - left);
            }else{
                countChar.remove(s.charAt(left++));
            }
        }
        return longSoFar;
    }

File: longest-substring-without-repeating-characters/LongestSubstringWithoutRepeatingCharacters.py
def longestSubStringWithoutRepeatingCharacters(string):
    left = 0
    right = 0
    countset = set()
    longest = 0
    while left < len(string) and right < len(string):
        if not string[right] in countset:
            countset.add(string[right])
            right = right + 1
            longest = max(longest, right - left)
        else:
            countset.remove(string[left])
            left = left + 1
    return longest

result = longestSubStringWithoutRepeatingCharacters("pwwkew")
print(result)

